containers/sets/thread_safe_expirable.go

```diff
@@ -6,8 +6,8 @@ import (
 )
 
 type threadSafeExpirableSet struct {
-	mutex      sync.RWMutex
-	underlying Set
+	sync.RWMutex
```

Comment:
What's the value of using type embedding here? The mutex is an implementation detail and shouldn't show up in the contract. Thoughts?

Reply:
Do you mean the package contract? it won't show in the contract anyway given that the whole struct is not exported, the whole thing is just a small refactor to get rid of the underlying object since it just seemed like it not adding much value.

Reply:
No, the type contract. For functions in the package, to call `threadSafeExpirableSet.Lock()` sounds a bit off; calling `threadSafeExpirableSet.mutex.Lock()` makes more sense since it's a mutex operation (rather than a set operation).
The change to remove `underlying` looks good; it actually made me question why I added it to start with (I thought I had a reason but I can't recall it :))

Reply:
Fair enough, I can add the mutex back

Reply:
Thank you!